docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,201 @@
+
+## Desired User Experience (UX)
+
+### Deployment and Upgrade Form
+
+When users reach the deployment form, the desired behavior is to have two different views, one for the basic form and another for an advanced deployment. That can be represented as two different tabs. The basic deployment should contain all the basic parameters while the advanced form will show the information as we do today. For the first version, changes made in the basic deployment form won't affect the advanced form and viceversa. Only when the user clicks on "submit" the current form information (either basic or advanced) will be sent to the backend.
+
+This is the proposed view for the basic deployment form:
+
+<img src="./img/deployment-form-basic.png" width="500px">
+
+And for the advanced tab:
+
+<img src="./img/deployment-form-adv.png" width="500px">
+
+*Note that the design may vary, the final proposal will be submitted in a different PR
+
+Once a user has deployed the application, when trying to upgrade, the basic form should be shown. The form will be pre-filled with the values of the previous version. When switching to a new version, those basic parameters should be maintained.
+
+If the user decides to select the advanced version of the form, we should represent the existing values so the user can copy-paste them to the new version. Ideally, we could represent a git-patch view to notify the user of the changes in the previous `values.yaml` but that is out of the scope for this design.
+
+## Basic Parameters Definition
+
+The basic parameters that an application supports should be defined per chart. Along with the different files of the application, we should include the information of these parameters as a different file. The proposed approach is to use the file: `values.schema.json`. This file will be a [JSON Schema](https://json-schema.org/). We have chosen this approach since it's also used by Helm v3 to validate the `values.yaml` of a chart ([link](https://github.com/helm/helm/pull/5350/)). Also, it's possible to autogenerate a form from a JSON schema as we do for the service catalog instances. We won't do so for the moment because we cannot be sure that if a `values.schema.json` exists, it represent all the relevants fields of a `values.yaml`. In any case, we won't autogenerate the form to give a better experience.
+
+This JSON Schema allows us to know the structure of the `values.yaml` file but, to know which parameters should be represented in the basic form, we need more information. In particular we need to map well known parameters, like "Disk size", to parameters in the `values.yaml`. For doing so, we can include a new file: `values.form.json`. Having those two pieces of information we are able to deliver the desired experience. See the next section for an example.
```

Comment:
My concern about this is that we're forcing Helm chart maintainers to introduce a new file. Before asking people to add this file, It would be nice if we can analyze several `values.yaml` files to find common patterns. If we find them, maybe this file is not required. 

This is just a comment. I know it's quite difficult to find pattern on these files 😅

Reply:
I guess that with the new file you mean the `values.form.json` (not the `values.schema.json` that is made for Helm v3 as a "linter").

It's true that it includes some overhead for the maintainers but this file is not that big so it's relatively easy to implement. We are also not targeting all the apps in the registries, covering just a few of them will help many users.

In any case I think there is no other option. We are not going to be able to know if the username is called `wordpressUsername`, `username` or `wpUsername` and it may change in the future (we don't have control over the development of that chart). If we try to auto-guess those parameters several things are very likely to happen:

 - We will miss paramaters that should be included because of a naming convention we are not aware.
 - We may include parameters that are not relevant.
 - We may parse the wrong parameters. If we are looking for the disk size for wordpress we may end up modifying the one for mariadb.

Reply:
I still think this could be part of the `values.schema.json` - just an annotation/field to identify fields which should be included. I'll comment on the other thread we had about that (which disappeared from the diff since your change).